runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -1096,7 +1096,7 @@ J9::ValuePropagation::constrainRecognizedMethod(TR::Node *node)
             }
 
          if (!canTransformIdentityArrayElementLoadStoreUseTypeHint &&
-             TR::Compiler->cls.isPrimitiveValueTypeClass(hintComponentClass))
+             TR::Compiler->cls.isArrayNullRestricted(comp(), typeHintClass)) //TODO: typeHintClass eventually should be the null-restricted array class
```

Comment:
I would suggest making this `TODO` into a `TODO-VALUETYPE`

Reply:
Addressed in [7bc96d5](https://github.com/eclipse-openj9/openj9/compare/9fd2f09e62d10c0d31fb534bace077e933045ec4..7bc96d56b54a0663b220ede073dc58ef195f2dbe)